lib/server/devicestatus.js

```diff
@@ -6,9 +6,23 @@ function storage (collection, ctx) {
   var ObjectID = require('mongodb').ObjectID;
 
   function create(obj, fn) {
-    if (! obj.hasOwnProperty('created_at')){
-      obj.created_at = (new Date()).toISOString();
+    
+    let d = new Date();
+
+    try {
+      if (obj.created_at) d = new Date(obj.created_at);
```

Comment:
wondering if we should use moment here, I think it can handle more date formats than the built in `new Date`

also would be nice to share this code with treatments

Reply:
Treatments has _almost_ the same code. I hadn't known about the preBolus treatment creation before reading the test and the treatment code. Wonder if it's actually in use anywhere...

Reply:
Not sure how often it's used, but it's displayed on the careportal form as Carb Time for Snack and Meal Bolus